### PR TITLE
Fixed squash and unsquash in the case of explicit dummies

### PR DESCRIPTION
**Pull request by msuruzhon**
This PR fixes an issue where only the coordinates of the first molecule were used for squashing and unsquashing, leading to numerical instabilities in the simulations.

### CODE REVIEW

File: python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py [hunk 1]
```diff
@@ -113,9 +113,29 @@ def _squash_molecule(molecule, explicit_dummies=False):
...
+            dummies=False,
+        )
+        atom_mapping1_common = _squashed_atom_mapping(
+            molecule,
+            is_lambda1=True,
+            explicit_dummies=explicit_dummies,
+            environment=False,
+            dummies=False,
+        )
+        assert set(atom_mapping0_common) == set(atom_mapping1_common)
```
Comment (xiki-tempula) [inline python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py]: Not good to use assert in non-test cases.

**Review by xiki-tempula — CHANGES_REQUESTED** [review]
Code looks good. Add a test

File: python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py (hunk 1 above)
Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/Align/_squash.py]: Asserts should be used for statements that should always be true and are not the user's fault right? This is the case here.

**Comment by msuruzhon** [comment]
Hey @xiki-tempula, the tests pass but I guess GitHub was not very fond of me force pushing: https://github.com/Exscientia/biosimspace/actions/runs/5715816597. Let me know if you are happy for me to merge thanks.

**Review by xiki-tempula — APPROVED** [review]
LGTM

**Pull request merged**